### PR TITLE
able to set proxy agent to cognito http client

### PR DESCRIPTION
**Pull request by richardzcode**
Be able to set "agent" property to cognito http client.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/src/Client.js [hunk 1]
```diff
@@ -26,13 +27,16 @@ export default class Client {
...
 
-    const options = {
-      headers,
-      method: 'POST',
-      mode: 'cors',
-      cache: 'no-cache',
-      body: JSON.stringify(params),
-    };
+    const options = Object.assign(
+      this.options,
```
Comment (elorzafe) [inline packages/amazon-cognito-identity-js/src/Client.js]: I think this should be the opposite way, `this.options` should be at the end

Comment (powerful23) [reply packages/amazon-cognito-identity-js/src/Client.js]: Agree

**Comment by powerful23** [comment]
@richardzcode can you make the change mentioned above and resolve the conflicts so we can merge this into the repo? Thanks.

**Comment by elorzafe** [comment]
@powerful23 what should we do with this pr?

**Comment by powerful23** [comment]
@richardzcode I am going to close this pr now. If you want to continue merging this, please reopen it.

**Pull request closed without merging**